### PR TITLE
Add option to reset password of agents to Admin

### PR DESCRIPTION
**Pull request by mukesh4139**
# Pull Request Template

## Description

Add ability to admin to send reset password for agents.
Covers #348 

Fixes # (issue)

## Type of change

Please delete options that are not relevant.
- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

I have tested this manually.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by mukesh4139** [comment]
@pranavrajs please help review.

Please let me know if I have not followed any convention of the Frontend App.

**Comment by mukesh4139** [comment]
Name: puma
Version: 4.3.0
Advisory: CVE-2019-16770
Criticality: High
URL: https://github.com/puma/puma/security/advisories/GHSA-7xx3-m584-x994
Description:

  A poorly-behaved client could use keepalive requests to monopolize Puma's
  reactor and create a denial of service attack.

  If more keepalive connections to Puma are opened than there are threads
  available, additional connections will wait permanently if the attacker sends
  requests frequently enough.

Solution: upgrade to ~> 3.12.2, >= 4.3.1

Vulnerabilities found!

@pranavrajs @sojan-official how can I fix this?

**Comment by sojan-official** [comment]
@mukesh4139 let's do a bundle update.

**Comment by sojan-official** [comment]
#352 will fix this.

File: app/javascript/dashboard/routes/dashboard/settings/agents/EditAgent.vue [hunk 1]
```diff
@@ -132,6 +143,22 @@ export default {
...
       });
     },
+    resetPassword() {
+      // Call resetPassword from Auth Service
+      Auth.resetPassword(this.agentCredentials)
+        .then(res => {
+          this.editAgentsApi.message = this.$t('AGENT_MGMT.EDIT.PASSWORD_RESET.SUCCESS_MESSAGE');
+          this.showAlert();
+          this.resetForm();
+          setTimeout(() => {
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/agents/EditAgent.vue]: Why do we use setTimeouts?

Comment (mukesh4139) [reply app/javascript/dashboard/routes/dashboard/settings/agents/EditAgent.vue]: This is to close the EditAgent popup.

File: app/javascript/dashboard/i18n/locale/en/agentMgmt.json [hunk 2]
```diff
@@ -83,6 +83,10 @@
       "API": {
         "SUCCESS_MESSAGE": "Agent updated successfully",
         "ERROR_MESSAGE": "Could not connect to Woot Server, Please try again later"
+      },
+      "PASSWORD_RESET": {
+        "SUCCESS_MESSAGE": "Agent password reset successfully",
```
Comment (pranavrajs) [inline app/javascript/dashboard/i18n/locale/en/agentMgmt.json]: "An email with reset password instructions has been sent to the agent"

File: app/javascript/dashboard/routes/dashboard/settings/agents/EditAgent.vue (hunk 1 above)
Comment (pranavrajs) [reply app/javascript/dashboard/routes/dashboard/settings/agents/EditAgent.vue]: @mukesh4139 I don't think we need to close the Popup on successful reset.

**Comment by pranavrajs** [comment]
@mukesh4139 Could you please post a screenshot/screencast if possible? That will make reviews easier.

**Comment by mukesh4139** [comment]
@pranavrajs you can view the recording here:
https://drive.google.com/file/d/1Iq9eJU5vZklY8knpwk2JCotbNIGzX1c9/view?usp=sharing

**Pull request merged**